tests/test_modeling_gpt_neo.py

```diff
@@ -0,0 +1,470 @@
+        self.assertListEqual(
+            [isinstance(iter_attentions, tuple) for iter_attentions in attentions], [True] * len(attentions)
+        )
+        self.assertEqual(len(attentions), (max_length - min_length) * num_beam_groups)
+        for idx, iter_attentions in enumerate(attentions):
+            tgt_len = min_length + idx
+            src_len = min_length + idx
+            global_expected_shape = (
+                batch_size * num_beam_groups,
+                config.num_attention_heads,
+                tgt_len,
+                src_len,
+            )
+
+            local_seq_len, block_len, windows = self._get_local_attn_seq_len_block_len_windows(
+                src_len, config.window_size
+            )
+            local_expected_shape = (
+                batch_size * num_beam_groups,
+                windows,
+                config.num_attention_heads,
+                block_len,
+                local_seq_len,
+            )
+
+            shapes = [layer_attention.shape for layer_attention in iter_attentions]
+
+            # every other layer is local attention layers
+            # so alternate between expected shapes
+            expected_shape = [
+                global_expected_shape if i % 2 == 0 else local_expected_shape for i, _ in enumerate(iter_attentions)
+            ]
+            # check attn size
+            self.assertListEqual(shapes, expected_shape)
+
+    # caching is not implemented
+    def test_beam_search_generate_dict_outputs_use_cache(self):
+        pass
+
+    def test_greedy_generate_dict_outputs_use_cache(self):
+        pass
+
+    # batch generation is not supported yet
+    @slow
+    def test_batch_generation(self):
+        pass
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in GPT_NEO_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = GPTNeoModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+@require_torch
+class GPTNeoModelLanguageGenerationTest(unittest.TestCase):
+    @slow
+    def test_lm_generate_gpt_neo(self):
+        for checkpointing in [True, False]:
+            model = GPTNeoForCausalLM.from_pretrained("eleutherai/gpt_neo_xl", gradient_checkpointing=checkpointing)
+            model.to(torch_device)
+            input_ids = torch.tensor([[464, 3290]], dtype=torch.long, device=torch_device)  # The dog
+            expected_output_ids = [
+                464,
+                3290,
+                12,
+                3380,
+                4866,
+                286,
+                262,
+                1492,
+                11,
+                543,
+                318,
+                257,
+                4947,
+                286,
+                27126,
+                416,
+                262,
+                2739,
+                1772,
+                11,
+            ]  # The dog-eared copy of the book, which is a collection of essays by the late author,
```

Comment:
Put the comment on the line above `expected_output_ids = [` so all tokens can fit on the same line. If they do not, add comments to ignore styling.